packages/lib/src/user-query.ts

```diff
@@ -74,34 +74,69 @@ export const askForString = async (message: string, options?: AskForStringOption
 	return await promptForString(message, updatedOptions) as string
 }
 
-export const askForInteger = async (message: string, min?: number, max?: number): Promise<number | undefined> => {
-	const value = (await inquirer.prompt({
-		// Since 'number' converts strings to NaN without warning or error before validate is
-		// called, we use `input` here and just do our own validation. (It's possible to do some
-		// error correction in `transformer` but you can't return an error.)
+export type AskForIntegerOptions = {
+	default?: DefaultValueOrFn<number>
+
+	/**
+	 * A validation function targeted at `inquirer`'s prompt method. Note that `inquirer` input is
+	 * always a string, so this method gets the raw `string` entered by the user (not a `number`).
+	 * For simple min/max validation, use `integerValidateFn`.
+	 */
+	validate?: ValidateFunction
+
+	helpText?: string
+}
+
+const promptForInteger = async (message: string, options: AskForIntegerOptions): Promise<number | undefined> => {
+	const buildValidateFunction = (): ValidateFunction | undefined => {
+		// When there is a validate method defined and we have help text, we have to allow '?'.
+		if (options.helpText && options.validate) {
+			return input => input === '?' || options.validate === undefined || options.validate(input)
```

Comment:
Since this is inside an if statement that requires `options.validate` to be truthy, why does the returned function include `options.validate === undefined`?

Reply:
Good eye but it is necessary to make TypeScript happy because it's actually *not* wrapped in that `if` anymore when it gets called. The `buildValidateFunction` function returns a new one-line function validate function for use later. I violated our rule about naming functions that return functions with `Fn` so I've fixed that though.